src/reducers/index.js

```diff
@@ -0,0 +1,253 @@
+/**
+ * TODO : Global state should be flattenized (use of normalizr?)
+ */
+
+import * as types from '../constants/ActionTypes';
+import { combineReducers } from 'redux';
+
+const visibilityFilter = (state = {
+            shopName: 'SHOW_ALL',
+            shopStat: 'SHOW_ALL'
+        }, action) => {
+    switch (action.type) {
+        case types.SET_VISIBILITY_FILTER_STORE_STAT:
+            return Object.assign({}, { shopName: state.shopName, shopStat: action.filter });
+         case types.SET_VISIBILITY_FILTER_STORE_NAME:
+            return Object.assign({}, { shopName: action.filter, shopStat: state.shopStat });
+        default:
+            return state;
+    }
+};
+
+
+const inventory = (state = {}, action) => {
```

Comment:
This reducer is a little bit cryptic, and gets even more complicated with the `alternInventory` function. 

Do you think you could propose a simpler reducer/state structure (without rewriting all the components depending on this reducer) ?

Reply:
Yes as mentioned in the response before it will be better to normalize the state : 

```json
{
"stores":{
    "byId":{
        "store1": {
            "id":"store1",
            "shoes" : ["shoes1", "shoes2", "shoes3" ...]
        },
        "store2": {
            "id":"store2",
            "shoes" : ["shoes1", "shoes2", "shoes3" ...]
        },
        ...
    },
    "allIds": ["store1", "store2", ... ]
},
"shoes": {
    "byId":{
        "shoe1" : {
            "id": "shoe1",
            "color": ["blue","red"]
        },
        "shoe2" : {
            "id": "shoe2",
            "color": ["blue","red"]
        }
        ...
    },
    "allIds" : ["shoe1", "shoe2", ...]
},
"inventory" :  {
    "byId":{
        "inventory1" : {
             "id": "inventory1",  
             "shoe" : "shoe1",
             "store" : "store1",
             "quantity" : 14
        },
        "inventory2" : {
             "id": "inventory2",  
             "shoe" : "shoe2",
             "store" : "store1",
             "quantity" : 10
        },
        "inventory3" : {
             "id": "inventory3",  
             "shoe" : "shoe3",
             "store" : "store1",
             "quantity" : 10
        },
        ...
    },
    "allIds" : ["inventory1","inventory2", "inventory3", ...]
}
}
```